### PR TITLE
 Add inertial parameters regressor method to KinDynComputations

### PR DESCRIPTION
**Pull request by traversaro**
In particular, add the inverseDynamicsInertialParametersRegressor that
computes the (6+nrOfDofs) \times (10*nrOfLinks) matrix  that maps the inertial parameters
to the part of inverse dynamics that depends on inertial parameters,
and the generalizedExternalForces that computes the contribution of external
forces to inverse dynamics (useful because those are not included in the
torques computed by the regressor).

I would still like to add some more docs, but nevertheless the code and the tests should be complete.

### CODE REVIEW

**Comment by traversaro** [comment]
With these commit, the `KinDynComputations` class should now have feature parity w.r.t. to the old DynamicsComputations class.

File: doc/dcTutorialCpp.md [hunk 1]
```diff
@@ -1,29 +1,36 @@
...
-    dynComp.loadRobotModelFromFile("model.urdf");
-    std::cout << "The loaded model has " << dynComp.getNrOfDegreesOfFreedom() 
-          << "internal degrees of freedom and " << dynComp.getNrOfLinks() 
+    ModelLoader mdlLoader;
+    bool ok = mdlLoader.loadModelFromFile(urdf_filename);
+
+    // Check that ok is actually true
+
+    KinDynComputations kinDynComp;
+    kinDynComp.loadRobotModel("model.urdf");
```
Comment (francesco-romano) [inline doc/dcTutorialCpp.md]: This should not work! 

`kinDynComp.loadRobotModel(mdlLoader.model());`

Comment (traversaro) [reply doc/dcTutorialCpp.md]: I have toooo many failed attempts to get good documentation around, I should just thrown them our. :D 
Anyway, nice catch!

File: src/high-level/include/iDynTree/KinDynComputations.h [hunk 2]
```diff
@@ -697,6 +697,39 @@ class KinDynComputations {
      */
     bool generalizedGravityForces(FreeFloatingGeneralizedTorques & generalizedGravityForces);
 
+    /**
+     * Compute the getNrOfDOFS()+6 vector of generalized external forces.
+     *
+     * The semantics of baseAcc, the base part of baseForceAndJointTorques
```
Comment (francesco-romano) [inline src/high-level/include/iDynTree/KinDynComputations.h]: IMHO using "variable name style", i.e. `baseAcc` and `baseForceAndJointTorques` in a function that does not accept these variables is a bit confusing.
Same for the `inverseDynamicsInertialParametersRegressor` method.

Comment (traversaro) [reply src/high-level/include/iDynTree/KinDynComputations.h]: Yes, that's a left over, thanks for catching that.

**Comment by traversaro** [comment]
I think I need to stop in trying to fix all the docs otherwise I will do that for one month. :  ) 
Feel free to review @francesco-romano 
@prashanthr05 I also tagged you for a review, but I think I will merge, feel free to comment also once the PR has been merged.

**Pull request merged**